File: 2868-the-wording-game/2868-the-wording-game.cpp
class Solution {
public:
    bool canAliceWin(vector<string>& a, vector<string>& b) {
        string w = a[0]; int i = 1, j = 0, k = 1;
        while(1) {
            if(k == 1) {
                if(j == b.size()) return true;
                if(w[0] == b[j][0] && w < b[j] || b[j][0] - w[0] == 1) {
                    w = b[j];
                    k ^= 1;
                }
                ++j;
            } else {
                if(i == a.size()) return false;
                if(w[0] == a[i][0] && w < a[i] || a[i][0] - w[0] == 1) {
                    w = a[i];
                    k ^= 1;
                }
                ++i;
            }
        }
    }
};

File: 2868-the-wording-game/2868-the-wording-game.rb
# @param {String[]} a
# @param {String[]} b
# @return {Boolean}
def can_alice_win(a, b)
    w, i, j, k= a[0], 1, 0, 1
    while true do
        if k == 1
            return true if j == b.length
            if w[0] == b[j][0] && w < b[j] || b[j][0].ord - w[0].ord == 1
                w = b[j]
                k ^= 1
            end
            j += 1
        else
            return false if i == a.length
            if w[0] == a[i][0] && w < a[i] || a[i][0].ord - w[0].ord == 1
                w = a[i]
                k ^= 1
            end
            i += 1
        end
    end
end